core/src/main/java/by/ilyin/core/controller/EmailCoreController.java

```diff
@@ -0,0 +1,67 @@
+package by.ilyin.core.controller;
+
+import by.ilyin.core.dto.ChangeEmailDTO;
+import by.ilyin.core.dto.RestoreAccountDTO;
+import by.ilyin.core.dto.SendEmailDTO;
+import by.ilyin.core.service.EmailService;
+import by.ilyin.core.util.validation.custom.ConsistentChangeEmailParameters;
+import by.ilyin.core.util.validation.custom.ValidEmailRepairData;
+import by.ilyin.core.util.validation.custom.ValidIdByUserExists;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api")
+@Validated
+public class EmailCoreController {
+
+    private final EmailService emailService;
+
+    @PostMapping("/email")
+    public ResponseEntity<Void> sendMail(@RequestBody SendEmailDTO sendEmailDTO) {
+        emailService.sendEmail(sendEmailDTO);
+        return ResponseEntity
+                .ok()
+                .build();
+    }
+
+    @PostMapping("/email/repairing")
+    public ResponseEntity<Void> repairEmail(@RequestBody @ValidEmailRepairData SendEmailDTO sendEmailDTO) {
+        emailService.repairEmail(sendEmailDTO);
+        return ResponseEntity
+                .ok()
+                .build();
+    }
+
+    @PutMapping("/restore/{uuid}")
+    public ResponseEntity<Void> restoreAccount(@PathVariable("uuid") String uuid,
+                                               @RequestBody RestoreAccountDTO restoreAccountDTO) {
+        emailService.restoreAccount(uuid, restoreAccountDTO);
+        return ResponseEntity
+                .ok()
+                .build();
+    }
+
+    @PostMapping("/profile/change-email")
+    @ConsistentChangeEmailParameters
+    public ResponseEntity<Void> changeEmail(@RequestParam("userId") Long userId,
```

Comment:
@YauhenIlyin Where did you find userId as RequestParam in ticket?

Reply:
we have a lot of users in the system. The user himself does not provide information about himself when requested.
on the web, we need to extract the userId and pass it to core in order to understand which email to change.

Reply:
@DzmitryKazakevich 

we have a lot of users in the system. The user himself does not provide information about himself when requested.
on the web, we need to extract the userId and pass it to core in order to understand which email to change.

Reply:
@DzmitryKazakevich 
this is the internal logic between web and core